exchange/utils.go

```diff
@@ -17,6 +17,28 @@ import (
 	"github.com/prebid/prebid-server/privacy/lmt"
 )
 
+func bidderToPrebidSChains(req *openrtb_ext.ExtRequest) map[string]*openrtb_ext.ExtRequestPrebidSChainSChain {
+	bidderToSChains := make(map[string]*openrtb_ext.ExtRequestPrebidSChainSChain)
+
+	if req.Prebid.SChains == nil || len(req.Prebid.SChains) == 0 {
+		return bidderToSChains
+	}
+
+	for _, schainWrapper := range req.Prebid.SChains {
+		if schainWrapper != nil && len(schainWrapper.Bidders) > 0 {
+			for _, bidder := range schainWrapper.Bidders {
+				if _, present := bidderToSChains[bidder]; present {
```

Comment:
This seems strange. If a bidder is in the list, and that bidder also has an entry in `bidderToSChains`, we delete that entry, otherwise we add an entry for the bidder into the map? This would mean if a bidder appears an even number of times, it will be absent from the map. If it appears an odd number of times, it will be in the map with the last `SChain` it was seen with. What is the logic supposed to be here?

Reply:
Good catch. I don't see any info in the spec or in the applicable github issues/PRs about how to handle the case where multiple non-wildcard schains are defined for a bidder. The java version appears to take the same approach of ignoring all schains for a bidder if multiple are defined for a bidder, which is where I got the idea from. I'm not a java expert but it looks like the java version has the same bug. I'll confirm that this is the expected behavior and update the logic accordingly.

Reply:
The team decided that this is a data error that should be detected during request validation preventing further processing.